src/components/Gauge.js

```diff
@@ -0,0 +1,263 @@
+    dataPoints: React.PropTypes.array,
+    defaultLabelText: React.PropTypes.string,
+    defaultLabelValue: React.PropTypes.string,
+    formatter: React.PropTypes.func,
+    height: React.PropTypes.number,
+    id: React.PropTypes.string,
+    opacity: React.PropTypes.number,
+    padAngle: React.PropTypes.number,
+    showBaseArc: React.PropTypes.bool,
+    showDataLabel: React.PropTypes.bool,
+    width: React.PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      activeOffset: 0,
+      arcWidth: 10,
+      baseArcColor: StyleConstants.Colors.BASE_ARC,
+      colors: [StyleConstants.Colors.PRIMARY].concat(d3.scale.category20().range()),
+      data: [],
+      dataPointColors: [StyleConstants.Colors.CHARCOAL].concat(d3.scale.category20b().range()),
+      dataPointRadius: 5,
+      dataPoints: [],
+      formatter (value) {
+        return value;
+      },
+      height: 150,
+      id: 'gauge',
+      opacity: 1,
+      padAngle: 0.02,
+      showBaseArc: true,
+      showDataLabel: true,
+      width: 150
+    };
+  },
+
+  getInitialState () {
+    return {};
+  },
+
+  componentWillMount () {
+    this._setupD3Functions(this.props);
+  },
+
+  componentWillReceiveProps (newProps) {
+    if (!_isEqual(this.props.data, newProps.data)) {
+      this._setupD3Functions(newProps);
+    }
+  },
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return !_isEqual(this.props, nextProps) || !_isEqual(this.state, nextState);
+  },
+
+  _setupD3Functions (props) {
+    const dataSets = props.data.map(item => {
+      return item.value;
+    });
+    const startAngle = (-135 * (Math.PI / 180));
+    const endAngle = (135 * (Math.PI / 180));
+    const pie = d3.layout.pie().sort(null).padAngle(props.padAngle).endAngle(endAngle);
+    const values = pie(dataSets);
+    const radius = Math.min(props.width, props.height) / 2;
+    const standardArc = d3.svg.arc().outerRadius(radius - props.activeOffset).innerRadius(radius - props.arcWidth);
+    const hoveredArc = d3.svg.arc().outerRadius(radius).innerRadius(radius - props.arcWidth);
+    const baseArc = d3.svg.arc().outerRadius(radius - props.activeOffset).innerRadius(radius - props.arcWidth).startAngle(startAngle).endAngle(endAngle);
+
+    this.setState({
+      baseArc,
+      endAngle,
+      hoveredArc,
+      pie,
+      radius,
+      standardArc,
+      values
+    });
+  },
+
+  _renderArcs () {
+    const firstSegment = -135 * (Math.PI / 180);
+    const endOfFirstSegment = -90 * (Math.PI / 180);
+    const secondSegment = -90 * (Math.PI / 180);
+    const endOfSecondSegment = -45 * (Math.PI / 180);
+    const thirdSegment = -45 * (Math.PI / 180);
+    const endOfThirdSegment = 0;
+    const fourthSegment = 0;
+    const endOfFourthSegment = 45 * (Math.PI / 180);
+    const fifthSegment = 45 * (Math.PI / 180);
+    const endOfFifthSegment = 90 * (Math.PI / 180);
+    const sixthSegment = 90 * (Math.PI / 180);
+    const endOfSixthSegment = 135 * (Math.PI / 180);
```

Comment:
There design only had 6 pieces, but you're right. I think this should be dynamic. I can do that--the math is pretty easy.